src/services/redisSessionStorage.ts

```diff
@@ -307,6 +308,14 @@ export default class RedisSessionStorage extends RedisStorageUtils
       .run();
   }
 
+  public async userHasLoginBlocked(
+    fiscalCode: FiscalCode
+  ): Promise<Either<Error, boolean>> {
+    return fromEither(
+      tryCatch(() => this.redisClient.exists(`USER-BLOCKED-${fiscalCode}`))
```

Comment:
@cloudify we have to implement a way to temporarely block users while the data deletion procedure is running.

we've chosen to put a flag 'block-user' into redis for each blocked CF. we can implement it either:
1) with a USER-BLOCKED-<CF> entry
2) with a USER-BLOCKED redis Set

what do you think about this approach? 
(2 will let us to enumerate blocked users without a full scan on redis keys)

Reply:
@gunzip `EXISTS` ([here](https://redis.io/commands/exists)) and `HGET` ([here](https://redis.io/commands/hget)) have the same complexity `O(1)`

Reply:
yes it's the enumeration complexity that changes (not the lookup). if you want to list all blocked users without a set you have to do a full scan of all redis keys (same if you want to clear the list of blocked users)